migrations/default/2021_07_08_000001_update_twill_default_users_table.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+use A17\Twill\Models\User;
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+class UpdateTwillDefaultUsersTable extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        $twillUsersTable = config('twill.users_table', 'twill_users');
+
+        if (Schema::hasTable($twillUsersTable)) {
+            Schema::table($twillUsersTable, function (Blueprint $table) {
+                $table->boolean('activated')->default(false);
+                $table->dateTime('last_login_at')->nullable();
+                $table->dateTime('registered_at')->nullable();
+                $table->boolean('require_new_password')->default(false);
+            });
+
+            $this->seedNewFields();
+        }
+    }
+
+    /**
+     * Reverse the migrations.
+     *
+     * @return void
+     */
+    public function down()
+    {
+        $twillUsersTable = config('twill.users_table', 'twill_users');
+
+        if (Schema::hasTable($twillUsersTable)) {
+            Schema::table($twillUsersTable, function (Blueprint $table) {
+                $table->dropColumn('activated');
+                $table->dropColumn('last_login_at');
+                $table->dropColumn('registered_at');
+                $table->dropColumn('require_new_password');
+            });
+        }
+    }
+
+    private function seedNewFields()
+    {
+        User::chunk(100, function ($users) {
+            foreach ($users as $user) {
+                if (!empty($user->password)) {
+                    $user->activated = true;
+                }
+
+                if (!empty($user->created_at)) {
+                    $user->registered_at = $user->created_at;
```

Comment:
@pboi20 i wonder if this should be set in the above condition too. If the user currently doesn't have a password, it is not considered registered. It makes me question the need for the activated boolean, do we actually need it on top of the registration date?

Reply:
Makes sense! Good question about the `activated` column, I will dig a bit deeper and confirm if it's actually needed. At a glance, it does feel somewhat redundant...